src/PrestaShopBundle/Form/Admin/Improve/International/Geolocation/GeolocationFormDataProvider.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Improve\International\Geolocation;
+
+use PrestaShop\PrestaShop\Adapter\Validate;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Form\FormDataProviderInterface;
+use PrestaShop\PrestaShop\Core\Geolocation\GeoLite\GeoLiteCityCheckerInterface;
+
+/**
+ * Class GeolocationFormDataProvider is responsible for handling geolocation form data
+ */
+final class GeolocationFormDataProvider implements FormDataProviderInterface
+{
+    /**
+     * @var DataConfigurationInterface
+     */
+    private $geolocationByIpAddressConfiguration;
+
+    /**
+     * @var DataConfigurationInterface
+     */
+    private $geolocationIpAddressWhitelistConfiguration;
+
+    /**
+     * @var DataConfigurationInterface
+     */
+    private $geolocationOptionsConfiguration;
+
+    /**
+     * @var GeoLiteCityCheckerInterface
+     */
+    private $geoLiteCityChecker;
+
+    /**
+     * @var Validate
+     */
+    private $validate;
+
+    /**
+     * @param DataConfigurationInterface $geolocationByIpAddressConfiguration
+     * @param DataConfigurationInterface $geolocationIpAddressWhitelistConfiguration
+     * @param DataConfigurationInterface $geolocationOptionsConfiguration
+     * @param GeoLiteCityCheckerInterface $geoLiteCityChecker
+     * @param Validate $validate
+     */
+    public function __construct(
+        DataConfigurationInterface $geolocationByIpAddressConfiguration,
+        DataConfigurationInterface $geolocationIpAddressWhitelistConfiguration,
+        DataConfigurationInterface $geolocationOptionsConfiguration,
+        GeoLiteCityCheckerInterface $geoLiteCityChecker,
+        Validate $validate
```

Comment:
Would you mind to create an empty interface called ValidatorInterface into src/Core/Validation and make the adapter implements it?

Reply:
and rename the validate property to validator